src/wasm_opt.rs

```diff
@@ -8,11 +8,23 @@ use std::{
 
 lazy_static! {
     static ref WASM_OPT_URL: String = {
+        let version = "110";
+
+        #[cfg(target_arch = "arm")]
+        let arch = "arm64";
+
+        #[cfg(target_arch = "x86_64")]
+        let arch = "x86_64";
+
+        #[cfg(target_os = "windows")]
+        let os = "windows";
+        #[cfg(target_os = "macos")]
+        let os = "macos";
+        #[cfg(target_os = "linux")]
+        let os = "linux";
```

Comment:
The problem with this is that if the target arch or os is different than anything you enumerated, it will fail to compile with an error message that will not be easy to understand for the user

https://doc.rust-lang.org/reference/conditional-compilation.html#target_arch